source/plugins/devices/wyapp/devices/raspberrypi/index.js

```diff
@@ -76,6 +76,38 @@ export function setup (options, imports, register)
 		run (project)
 		{
 			project;
+		},
+
+		/**
+		 * Modidify the project before deploy
+		 * @param {Project} project - the project
+		 */
+		deploy (project)
+		{
+			let dockerfile = null;
+			// TODO add here the dockerfile
+			if(project.language === 'nodejs')
+			{
+				dockerfile = 'FROM balenalib/raspberrypi3-debian-node:latest';
+			}
+			else{
+				dockerfile = 'FROM balenalib/raspberrypi3-debian-python:latest';
+			}
```

Comment:
This should be delegated to the device plugin, each device knows its images.

Reply:
The code is already in the raspberry pi plugin, and the images depend on the used language. In this case shouldn't I delegate this to the language plugin?

Reply:
My mistake.